app/models/concerns/language_and_schedule_references.rb

```diff
@@ -7,5 +7,17 @@ module LanguageAndScheduleReferences
 
     has_many :schedules, as: :scheduleable, dependent: :destroy
     accepts_nested_attributes_for :schedules
+
+    before_validation :build_schedules
+    validate :validate_schedule_count
+
+    def validate_schedule_count
+      errors.add(:schedules, 'too many') if schedules.size > 21
+      errors.add(:schedules, 'not enough') if schedules.size < 21
```

Comment:
@toupeira But how am I supposed to get this thing green with rubocop? What you suggest results in `Metrics/AbcSize: [15.17/15]`

```bash
language_and_schedule_references.rb:14:5: C: Metrics/AbcSize: Assignment Branch Condition size for validate_schedule_count is too high. [15.17/15]
    def validate_schedule_count
```

```ruby
def validate_schedule_count
  schedule_supposed_size = Schedule.days.size * Schedule.times.size
  if schedules.size < schedule_supposed_size
    errors.add(:schedules, 'not enough')
  elsif schedules.size > schedule_supposed_size
    errors.add(:schedules, 'too many')
  end
end
```